[🐞] Cannot find module '/@builder.io/qwik/build'

### Which component is affected?

Qwik Runtime

### Describe the bug

I am unable to run any test, I get the following error when calling `createDOM()`:

```text
Error: Cannot find module '/@builder.io/qwik/build' imported from '<absolute_path_to>/node_modules/.pnpm/@builder.io+qwik@1.1.5_undici@5.22.1/node_modules/@builder.io/qwik/core.mjs'.

- If you rely on tsconfig.json to resolve modules, please install "vite-tsconfig-paths" plugin to handle module resolution.
 - Make sure you don't have relative aliases in your Vitest config. Use absolute paths instead. Read more: https://vitest.dev/guide/common-errors
 ❯ node_modules/.pnpm/@builder.io+qwik@1.1.5_undici@5.22.1/node_modules/@builder.io/qwik/core.mjs:1:31
      1| /**
      2|  * @license
      3|  * @builder.io/qwik 1.1.5
       |               ^
      4|  * Copyright Builder.io, Inc. All Rights Reserved.
      5|  * Use of this source code is governed by an MIT-style license that can be
 ❯ tests/MyComponent.spec.tsx:1:31
```

### Reproduction

https://stackblitz.com/github/devatina11yb/qwik-testing

### Steps to reproduce

StackBlitz seems to think I am using npm even though I am actually using pnpm. You will probably need to delete both `package-lock.json` and `pnpm-lock.yaml` and then run `pnpm install`.

1. `pnpm install`
2. `pnpm run test`

### System Info

```shell
@builder.io/qwik: 1.1.5
typescript: 5.1.3
undici: 5.22.1
vite: 4.3.9
vitest: 0.32.0

Windows: 11
pnpm: 8.6.1
```


### Additional Information

I also tried with npm and I get the exact same error so it doesn't seem to be related to pnpm.

--- a/packages/qwik/src/optimizer/src/plugins/plugin.ts
+++ b/packages/qwik/src/optimizer/src/plugins/plugin.ts
@@ -423,7 +423,7 @@ export function createPlugin(optimizerOptions: OptimizerOptions = {}) {
       };
     }
 
-    if (opts.resolveQwikBuild && id === QWIK_BUILD_ID) {
+    if (opts.resolveQwikBuild && id.endsWith(QWIK_BUILD_ID)) {
       log(`resolveId()`, 'Resolved', QWIK_BUILD_ID);
       return {
         id: normalizePath(getPath().resolve(opts.rootDir, QWIK_BUILD_ID)),
@@ -570,7 +570,7 @@ export function createPlugin(optimizerOptions: OptimizerOptions = {}) {
       TRANSFORM_REGEX.test(pathId) ||
       insideRoots(ext, dir, opts.srcDir, opts.vendorRoots)
     ) {
-      const strip = opts.target === 'client' || opts.target === 'ssr';
+      const strip = opts.target !== 'lib';
       const normalizedID = normalizePath(pathId);
       log(`transform()`, 'Transforming', pathId);
 

--- a/starters/adapters/deno/src/entry.deno.ts
+++ b/starters/adapters/deno/src/entry.deno.ts
@@ -41,7 +41,7 @@ async function serveHttp(conn: any) {
   // Each request sent over the HTTP connection will be yielded as an
   // async iterator from the HTTP connection.
   for await (const requestEvent of httpConn) {
-    const staticResponse = await staticFile(requestEvent.request);
+    const staticResponse = await staticFile(requestEvent.request, conn);
     if (staticResponse) {
       // Serve static file
       requestEvent.respondWith(staticResponse);

